util/src/main/java/io/grpc/util/SpiffeIdParserImpl.java

```diff
@@ -0,0 +1,85 @@
+package io.grpc.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class SpiffeIdParserImpl implements SpiffeIdParser {
+
+  private final static String PREFIX = "spiffe://";
+
+  @Override
+  public SpiffeIdInfo parse(String uri) {
+    validateFormat(uri);
+    String domainAndPath = uri.substring(PREFIX.length());
+    String trustDomain;
+    String path = "";
```

Comment:
nit: Let's be consistent about whether or not we default initialize to the empty string or not. (I know this is also done in the reference code, please feel free to fix it there as well.)

Reply:
That's a Java thing - trustDomain is initialized in both `if` branches below, but path in only one, that's why we have to do it when we define it. I can also move path initialization to another branch, but not sure it it'll pass the styling test

Reply:
Moved empty string assignment to the first branch